### PR TITLE
style: migrate to ruff

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
Follow up of #558 

Request for Review: @epwalsh

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -133,9 +133,9 @@ When you're ready to contribute code to address an open issue, please follow the
         isort .
         black .
 
-    Our CI also uses [`flake8`](https://github.com/allenai/tango/tree/main/tests) to lint the code base and [`mypy`](http://mypy-lang.org/) for type-checking. You should run both of these next with
+    Our CI also uses [`ruff`](https://github.com/allenai/tango/tree/main/tests) to lint the code base and [`mypy`](http://mypy-lang.org/) for type-checking. You should run both of these next with
```
Comment (epwalsh) [inline .github/CONTRIBUTING.md:136]: Let's just link to ruff instead
```suggestion
    Our CI also uses [`ruff`](https://github.com/charliermarsh/ruff) to lint the code base and [`mypy`](http://mypy-lang.org/) for type-checking. You should run both of these next with
```

Comment (epwalsh) [inline .github/CONTRIBUTING.md:136]: Let's just link to ruff instead
```suggestion
    Our CI also uses [`ruff`](https://github.com/charliermarsh/ruff) to lint the code base and [`mypy`](http://mypy-lang.org/) for type-checking. You should run both of these next with
```

File: pyproject.toml [hunk 2]
```diff
@@ -156,6 +156,23 @@ exclude = '''
 profile = "black"
 multi_line_output = 3
 
+[tool.ruff]
+line-length = 115
+select = ["E"]
+ignore = ["E203"]
```
Comment (epwalsh) [inline pyproject.toml:162]: Ruff is complaining about this ignore code

**Review by epwalsh — CHANGES_REQUESTED** [review]
Thanks @SauravMaheshkar! Looks like the ruff configuration needs a fix

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (SauravMaheshkar) [reply .github/CONTRIBUTING.md]: Addressed in 980da4e1bc464bdfbf01dcfeacdba3c3b2045122

File: pyproject.toml (hunk 2 above)
Comment (SauravMaheshkar) [reply pyproject.toml]: Addressed in 75156c798341938de94e1e2957120b9603d116a8

**Comment by SauravMaheshkar** [comment]
Are the failing tests related to this PR ??

**Review by epwalsh — APPROVED** [review]
Thank you!!

**Pull request merged**